tst/PeerConnectionFunctionalityTest.cpp

```diff
@@ -714,6 +758,16 @@ TEST_F(PeerConnectionFunctionalityTest, noLostFramesAfterConnected)
         THREAD_SLEEP(HUNDREDS_OF_NANOS_IN_A_MILLISECOND);
     }
 
+    this->lock.lock();
```

Comment:
We should utilize try/catch blocks to guarantee an unlock. Thread join can throw an exception.

Reply:
Might also want to looked into timed mutex for timed lock with timeouts